app/lib/smaxtec_api.rb

```diff
@@ -5,21 +5,46 @@ class SmaxtecApi
   SMAXTEC_API_PASSWORD = Rails.application.secrets.smaxtec_api_password
   SMAXTEC_API_BASE_URL = 'https://api-staging.smaxtec.com/api/v1'
 
-  def get_temperature
+  def get_sensor_readings
+    #define animal_id (smaxtec api identifier) for cows: topic_id => animal_id
+    animal_id = { 1 => '5722099ea80a5f54c631513d' }
+
+    # Assuming 1 report = 1 cow, otherwise get just the specified reports
+    Report.all.each do |report|
+      if animal_id[report.id]
+        sensors = Sensor.where(report_id: report.id, smaxtec_sensor: true)
+        if sensors
+          sensors.each do |sensor|
+            # temperature sensor
+            if sensor.sensor_type_id == 1
+              # get temperature data from smaxtec api
+              temperature = get_temperature(animal_id[report.id])
+
+              if temperature
+                # create new sensor readig for sensor
+                # compare temperature with latest sensor reading and only create new sensor reading if value has changed?
+                sensor_reading = Sensor::Reading.new(sensor_id: sensor.id, calibrated_value: temperature, uncalibrated_value: temperature)
```

Comment:
You mean unique Id for each sensor reading to avoid multiple entries for one sensor reading? I don't think so, but I will have to look at the API again.